tokio-sync/src/semaphore.rs

```diff
@@ -874,6 +880,7 @@ impl WaiterNode {
         NodeState::store(&self.state, Idle, Relaxed);
     }
 
+    #[allow(clippy::wrong_self_convention)]
```

Comment:
This can probably be `self: Arc<WaiterNode>` now?

Reply:
I did this in https://github.com/tokio-rs/tokio/pull/1335/commits/d63fc9c6030e452115f589fb9036a00e189e8976. But current `wrong_self_convention` seems to allow only `self: Self`.

Reply:
I re-allowed this lint in https://github.com/tokio-rs/tokio/pull/1335/commits/b5c8e69eb43c418cc75c528e6c9419e377285cee and filed https://github.com/rust-lang/rust-clippy/issues/4293.